rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,146 @@
+# Stop processing transactions after a certain slot
+
+This PR describes the feature to stop processing transactions after a certain
+slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks.
+
+In this RFC, the procedure to stop processing any new transactions after a
+certain slot is described. This is, any blocks produced after that slot will
+include no transaction at all.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. We want this to happen by defining a
+point in time (slot) where the network stops accepting transactions a keeps
+producing blocks free of transactions until it reaches a consensus state.
```

Comment:
Presumably the 'consensus state' should be a stabilised ledger? If so, don't we need to ensure that
* coinbases are 0
* fee-transfers are 0 (i.e. snark work fees are not paid)?
If not, please specify the details of what the stabilisation will look like, or link to a relevant document.

Reply:
Correct. I tried to clarify that.